luvit/luvi
# move all CI to Actions, overhaul build system, upgrade ancient dependencies

This is a large set of changes that accomplishes a few different goals all at the same time.

With these changes, we:

- luvi now correctly compiles on MinGW
- update from the unmaintained pcre to pcre2
- update to openssl 3.1
- update to latest lua-openssl release
- update to latest luv release
- update to latest zlib release
- update from cmake 2.x to cmake 3.15, and start using newer cmake features and clean up the CMakeLists.txt
- will provide binaries for more platforms
  - adding support for Darwin arm64, Linux aarch64, and Windows mingw
  - provide both Lua 5.4 and Luajit binaries for all targets
- switch await from holy-build-box, which has dropped support for 32-bit Linux 
  - it is being replaced with the python manylinux project
  - also docker based, but is actively maintained for a much wider range of architectures
  
Alongside this change, all CI is now handled on Github Actions instead of being split between Actions and appveyor. 

The makefile and make.bat have been moved into a single make *and* nmake compatible makefile. While it makes the makefile slightly more complicated, I felt that the distinction was necessary now that mingw builds are possible and make.bat would always attempt a msvc build.

The cmake Find\* modules have all been cleaned up and commented.

All precompiled lua files are now precompiled using luac.lua instead of luajit's object output, it was causing issues on arm64 Darwin and it's much more simple to handle PUC and Luajit the same here.

The luv package has also been added to package.preload as `luv` in order to be more compatible with what luarocks packages may expect.
